zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -144,6 +182,22 @@ object Cookie {
     case object None   extends SameSite { def asString = "None"   }
   }
 
+  def decodeResponseSignedCookie(headerValue: String, secret: String): Option[Cookie] = {
```

Comment:
I think we can do better here. Instead of adding a new function to decode, it would be easier, to use it like this - `decodeResponseCookie(message, Some("secret"))`.

```scala
def decodeResponseCookie(message: String, secret: Option[String] = None): Option[Cookie] = ...
```

The logic for this code should also move to `unsafeDecodeResponseCookie`. That we have all the decoding logic at one place.

```scala
def unsafeDecodeResponseCookie(headerValue: String, secret: String = null): Cookie = ???
```

Basic idea is to use `null` in unsafe code. This helps with performance.